repo/blob/gcs/gcs_pit.go

```diff
@@ -161,7 +145,7 @@ func maybePointInTimeStore(ctx context.Context, gcs *gcsStorage, pointInTime *ti
 		return nil, errors.Wrapf(err, "could not get determine if bucket '%s' supports versioning", gcs.BucketName)
 	}
 
-	if !attrs.VersioningEnabled {
+	if !attrs.VersioningEnabled || attrs.ObjectRetentionMode != "Enabled" {
 		return nil, errors.Errorf("cannot create point-in-time view for non-versioned bucket '%s'", gcs.BucketName)
```

Comment:
A separate check would be better. That way you can highlight that ObjectRetention is the problem in the err